src/aggregator/aggregator/follower_flush_mgr_test.go

```diff
@@ -131,6 +132,56 @@ func TestFollowerFlushManagerCanNotLeadForwardedFlushWindowsNotEnded(t *testing.
 	require.False(t, mgr.CanLead())
 }
 
+func TestFollowerFlushManagerCanLeadNotFlushed(t *testing.T) {
+	ctrl := gomock.NewController(t)
+	defer ctrl.Finish()
+
+	doneCh := make(chan struct{})
+	electionManager := NewMockElectionManager(ctrl)
+	electionManager.EXPECT().IsCampaigning().Return(true).AnyTimes()
+	opts := NewFlushManagerOptions().SetElectionManager(electionManager)
+	mgr := newFollowerFlushManager(doneCh, opts).(*followerFlushManager)
+	testFlushTimes := &schema.ShardSetFlushTimes{
+		ByShard: map[uint32]*schema.ShardFlushTimes{
+			0: &schema.ShardFlushTimes{
+				StandardByResolution: map[int64]int64{
+					10000000000:  0, // 10s
+					60000000000:  0, // 1m
+					600000000000: 0, // 10m
+				},
+				ForwardedByResolution: map[int64]*schema.ForwardedFlushTimesForResolution{
+					// 10s
+					10000000000: &schema.ForwardedFlushTimesForResolution{
+						ByNumForwardedTimes: map[int32]int64{
+							1: 0,
+						},
+					},
+					// 1 m
+					60000000000: &schema.ForwardedFlushTimesForResolution{
+						ByNumForwardedTimes: map[int32]int64{
+							1: 0,
+						},
+					},
+					// 10m
+					600000000000: &schema.ForwardedFlushTimesForResolution{
+						ByNumForwardedTimes: map[int32]int64{
+							1: 0,
+						},
+					},
+				},
+			},
+		},
+	}
+
+	mgr.processed = testFlushTimes
+
+	mgr.openedAt = time.Now().Add(-9 * time.Minute)
```

Comment:
Nit: let's do these as separate test cases (could table test it) instead of multiple asserts in one case.

Reply:
Table tests are less obvious to read and debug.
I encapsulated a test logic in a function and added multiple `t.Run()` as separate test cases.